### PR TITLE
Plugin-Update to Mautic3

### PR DESCRIPTION
**Pull request by luguenth**
User-Facing changes:
The Callback-URL has to be altered. The new Callback-URL is ending in "Auth0" instead of "Auth0Auth"

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,5 @@
-Auth0 for Mautic
-================
+Auth0 for Mautic 3
```
Comment (flossels) [inline README.md:1]: Please do not harden the version number here as this plugin is in older versions still available for versions prior to Mautic 3.

File: README.md [hunk 2]
```diff
@@ -90,3 +90,8 @@ You can configure the mapping (Auth0 data -> Mautic User data) in the configurat
 </table>
 
 Use dot syntax to access arrays (i.e. `user_metadata.login_name`).
+
+## Update from Mautic 2.x.x to Mautic 3.x.x
+When Updating the Plugin, you have to make sure to change the Callback-URL(s) in Auth0.
```
Comment (flossels) [inline README.md:116]: You can simplify this lines by just writing:
When updating the plugin, please make sure to change the callback URL from `../s/sso_login/Auth0Auth` to `../s/sso_login/Auth0` in your Auth0 application settings.

File: Config/config.php [hunk 3]
```diff
@@ -27,14 +26,36 @@
                 'class' => \MauticPlugin\MauticAuth0Bundle\EventListener\ConfigSubscriber::class,
             ],
         ],
+        'integrations' => [
+            'mautic.integration.auth0' => [
+                'class' => \MauticPlugin\MauticAuth0Bundle\Integration\Auth0Integration::class,
+                'arguments' => [
```
Comment (flossels) [inline Config/config.php:32]: Are all these arguments really necessary?

File: Integration/Auth0Integration.php [hunk 4]
```diff
@@ -37,27 +31,19 @@ class Auth0AuthIntegration extends AbstractSsoServiceIntegration
...
     }
 
     /**
+     * Return's authentication method such as oauth2, oauth1a, key, etc.
+     *
      * @return string
      */
     public function getAuthenticationType()
     {
+        // Just use none for now and I'll build in "basic" later
```
Comment (flossels) [inline Integration/Auth0Integration.php:46]: ???

File: Integration/Auth0Integration.php [hunk 5]
```diff
@@ -109,6 +95,22 @@ public function setUserProvider(UserProvider $userProvider)
         $this->userProvider = $userProvider;
     }
 
+
+    /**
+     * Set the callback URL to sso_login.
+     */
+    public function getAuthCallbackUrl()
```
Comment (flossels) [inline Integration/Auth0Integration.php:102]: This can be archived by using `sprintf()` only.

**Review by flossels — COMMENTED** [review]
Hi Lukas,

there are some things you should change before I will merge this PR.

**Review by flossels — CHANGES_REQUESTED** [review]
request changes.

**Review by flossels — CHANGES_REQUESTED** [review]
There is an issue with the file permissions which are partially changed from 0644 to 0755

**Comment by flossels** [comment]
@luguenth any news on this?

File: Config/config.php (hunk 3 above)
Comment (luguenth) [reply Config/config.php:32]: yes unfortunately :cry:

**Review by flossels — APPROVED** [review]

**Pull request merged**